### PR TITLE
Add pipeline action for integration tests

### PR DESCRIPTION
**Pull request by Dannyps**
# Readiness checklist

- [ ] I added/updated unit tests.
- [x] I added/updated integration tests.
- [x] I ensured that the PR title is good enough for the changelog.
- [x] I labeled the PR.

### CODE REVIEW

**Comment by Dannyps** [comment]
I realised now someone fixed the AdminUi tests. It was a good exercise, too bad it was needless 😭

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -189,3 +189,23 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install script dependencies
+        run: npm install --prefix ./.ci
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: "7.0.x"
+      - name: Build project
+        run: dotnet restore "Backbone.sln" && dotnet build  /property:WarningLevel=0 --no-restore "Backbone.sln"
```
Comment (tnotheis) [inline .github/workflows/test.yml:205]: Why do you restore and build the solution? This should be done by the Dockerfiles during docker compose up.

Btw, you should probably use Dockerfile instead of Dockerfile.debug in case of the Consumer API

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -189,3 +189,23 @@ jobs:
...
+      - name: Run Consumer API and Admin Ui/API
+        run: docker compose -f ./docker-compose/docker-compose.test.yml up consumer-api admin-ui seed-mssql seed-client -d
+      - name: Run tests
+        run: dotnet test --no-restore --no-build "Backbone.sln" --filter "Category=Integration&TestCategory!~ignore"
+      - name: Stop Consumer API
+        run: docker compose -f ./docker-compose/docker-compose.test.yml down
```
Comment (tnotheis) [inline .github/workflows/test.yml:211]: Extract these commands into a zx script, so that everyone can run the tests manually on their PC.

Same for the Consumer API commands

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (Dannyps) [reply .github/workflows/test.yml:205]: That's something I hadn't done initially, but it turns out that building the solution inside Docker makes it impossible to run the tests on the host machine later. See https://github.com/nmshd/backbone/actions/runs/5530508228/jobs/10090204253#step:7:12

File: .github/workflows/test.yml (hunk 2 above)
Comment (Dannyps) [reply .github/workflows/test.yml]: > Same for the Consumer API commands

Are you referring to the transport tests?

Comment (tnotheis) [reply .github/workflows/test.yml]: Oops. Yes, of course I do :)

File: docker-compose/docker-compose.test.yml [hunk 3]
```diff
@@ -27,17 +27,24 @@ services:
...
+        condition: service_started
+      rabbitmq:
+        condition: service_started
+      seed-mssql:
+        condition: service_completed_successfully
+      azure-storage-emulator:
+        condition: service_started
+    configs:
+      - source: ConsumerApiConfig
+        target: app/appsettings.json
```
Comment (tnotheis) [inline docker-compose/docker-compose.test.yml:40]: Is this really necessary? Because this file should actually be copied into the output directory during the build process. If it doesn't there is something wrong

File: docker-compose/docker-compose.test.yml [hunk 4]
```diff
@@ -127,6 +137,10 @@ services:
 configs:
   AdminUiOverrideConfig:
     file: ../AdminUi/src/AdminUi/appsettings.override.json
+  ConsumerApiConfig:
+    file: ../ConsumerApi/appsettings.json
```
Comment (tnotheis) [inline docker-compose/docker-compose.test.yml:141]: Remember to remove this if it is not necessary (see comment above)

File: docker-compose/docker-compose.test.yml [hunk 5]
```diff
@@ -26,16 +27,47 @@ services:
...
+      azure-storage-emulator:
+        condition: service_started
+    configs:
+      - source: ConsumerApiConfig
+        target: app/appsettings.json
+      - source: ConsumerApiOverrideConfig
+        target: app/appsettings.override.json
     healthcheck:
-      test: ["CMD", "curl", "-f", "http://localhost/health"]
+      test: ["CMD", "wget", "http://localhost/health"]
```
Comment (tnotheis) [inline docker-compose/docker-compose.test.yml:44]: Why did you remove the `start_period` from this healthcheck?

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: docker-compose/docker-compose.test.yml (hunk 3 above)
Comment (Dannyps) [reply docker-compose/docker-compose.test.yml]: image
It seems the `appsettings.override.json` file is missing. How should I proceed?

Comment (tnotheis) [reply docker-compose/docker-compose.test.yml]: My comment was only for appsettings.json.

File: docker-compose/docker-compose.test.yml (hunk 5 above)
Comment (Dannyps) [reply docker-compose/docker-compose.test.yml:42]: `start_period` is the time Docker will wait before starting the healthchecks.
It came to my attention that in some scenarios (when testing locally and the consumer api is already built) the _booting up_ of the consumer api is rather quick. Moreover, the value we had before was more random than anything.

So, by removing it, we're performing healthchecks from the moment the container starts running. Considering it doesn't harm the loading process, and it's not a heavy operation, I figured it best to leave this way.

Comment (tnotheis) [reply docker-compose/docker-compose.test.yml:42]: I probably would have configured a 5-10s waiting time. But you're right, it probably doesn't harm.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -23,6 +23,49 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install script dependencies
+        run: npm install --prefix ./.ci
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: "7.0.x"
```
Comment (tnotheis) [inline .github/workflows/test.yml:37]: I think you don't need dotnet, now that you don't run dotnet build/restore anymore

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -23,6 +23,49 @@ jobs:
...
+      - name: Run integration tests
+        run: ./.ci/integration-test.ts
+
+  cns-transport-test:
+    name: Run CNS-Transport Tests
```
Comment (tnotheis) [inline .github/workflows/test.yml:42]:
```suggestion
    name: Run cns-transport Tests
```

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — APPROVED** [review]
